### PR TITLE
Fixes #13821 - Show different types of input depending on the param type

### PR DESCRIPTION
**Pull request by sean797**
One problem with this is when the select box is disabled it's a slightly different grey :( Any pointers where this is defined would be useful if you don't mind? I haven't had any luck finding it.

screenshot from 2016-02-20 13-33-53

### CODE REVIEW

File: app/helpers/common_parameters_helper.rb [hunk 1]
```diff
@@ -29,9 +30,15 @@ def parameter_value_content(id, value, options)
     if lookup_key.present? && lookup_key.hidden_value?
       field = password_field_tag(id, value, option_hash)
     else
-      field = text_area_tag(id, value, option_hash)
+      case lookup_key_type
```
Comment (dLobatog) [inline app/helpers/common_parameters_helper.rb]: Please extract this to a separate method, `lookup_key_field` or something similar

File: app/helpers/lookup_keys_helper.rb [hunk 2]
```diff
@@ -75,7 +75,8 @@ def lookup_key_with_diagnostic(obj, lookup_key, lookup_value)
       :name => "#{lookup_value_name_prefix(lookup_key.id)}[value]",
       :disabled => !lookup_key.overridden?(obj) || lookup_value.use_puppet_default,
       :inherited_value => inherited_value,
-      :lookup_key => lookup_key)
+      :lookup_key => lookup_key,
+      :lookup_key_type => lookup_key.key_type)
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb:80]: No need to pass yet another parameter, just call `.key_type` on options[:lookup_key] on the receiver

File: app/views/puppetclasses/_class_parameters.html.erb [hunk 3]
```diff
@@ -23,8 +23,8 @@
       <div class="input-group">
         <%= lookup_key_with_diagnostic(obj, lookup_key, lookup_value) %>
         <span class="input-group-btn">
-          <%= hidden_toggle(lookup_key.hidden_value?, 'font', 'font', true) %>
-          <%= fullscreen_button("$(this).parent().prev()") %>
+          <%= hidden_toggle(lookup_key.hidden_value?, 'font', 'font', true) if lookup_key.hidden_value? %>
+          <%= fullscreen_button("$(this).parent().prev()") if lookup_key.key_type != "boolean" %>
```
Comment (dLobatog) [inline app/views/puppetclasses/_class_parameters.html.erb]: Use `unless` instead of `if negative_condition`

**Comment by dLobatog** [comment]
@sean797 I don't see a different kind of grey - it might be your windows manager, not CSS :smile: 

screenshot from 2016-02-22 09-48-08

Notice integration tests are broken because they look for 'textarea' in all cases, and this helper changes that.

Thanks for the patch, it makes params much easier to use!

File: app/helpers/lookup_keys_helper.rb (hunk 2 above)
Comment (sean797) [inline app/helpers/lookup_keys_helper.rb:80]: Yes, i thought that should have worked! But the tests failed with undefined method error - http://ci.theforeman.org/job/test_develop_pr_core/database=postgresql,label=fast,ruby=2.1/8187/

I'll try and work it out, Thanks @dLobatog!

Comment (sean797) [inline app/helpers/lookup_keys_helper.rb:80]: I've changed it like you said but as you'll see the tests are failing, I think we need this how I originally had it? Similar to https://github.com/theforeman/foreman/blob/develop/app/models/lookup_value.rb#L55-L56

**Comment by dLobatog** [comment]
Merged as ccdf25219a3c9493d3404f03a66c740c201bb30f, thanks @sean797!

**Pull request closed without merging**